manifests/config.pp

```diff
@@ -45,5 +45,13 @@
     'obey pam restrictions':    value => $samba::obey_pam_restrictions;
   }
 
+  $samba::idmap_config.each | $idmap_domain, $idmap_options | {
+    $idmap_options.each | $idmap_option, $idmap_value | {
+      samba::option { "idmap config ${idmap_domain}: ${idmap_option}":
```

Comment:
I think there should be a space before the colon here.  According the the smb.conf man page, all of the 'idmap config' examples contain spaces before and after the colon.

```

                idmap config * : backend        = tdb
                idmap config * : range          = 1000000-1999999

                idmap config MAIN : backend     = rid
                idmap config MAIN : range       = 5000000-5999999

                idmap config TRUSTED : backend  = rid
                idmap config TRUSTED : range    = 6000000-6999999
```